test/data/helm-chart/template.yaml

```diff
@@ -553,6 +401,275 @@ spec:
+                --namespace=${NAMESPACE} \
+                --values="$HELM_VALUES" \
+                developer-hub \
+                developer-hub/developer-hub; then
+                echo "ERROR while installing chart!"
+                exit 1
+              fi
+              echo "OK"
+        
+              echo -n "* Waiting for route: "
+              until kubectl get route "developer-hub" -o name >/dev/null ; do
+                echo -n "."
+                sleep 3
+              done
+              HOSTNAME="$(kubectl get routes "developer-hub" -o jsonpath="{.spec.host}")"
+              echo -n "."
+              if [ "$(kubectl get secret "$CHART-developer-hub-secret" -o name --ignore-not-found | wc -l)" = "0" ]; then
+                kubectl create secret generic "$CHART-developer-hub-secret" \
+                  --from-literal="hostname=$HOSTNAME" >/dev/null
+              fi
+              echo "OK"
+        
+              echo -n "* Updating app-config.yaml: "
+              kubectl get configmap developer-hub-app-config -o yaml > developer-hub-app-config.current.yaml
+              yq '.data.["app-config.yaml"]' developer-hub-app-config.current.yaml > app-config.yaml
+              touch app-config-update.yaml
+              echo -n "."
+        
+              # Set the base URL
+              URL="https://$HOSTNAME"
+              yq -i ".app.baseUrl = \"$URL\" | .backend.baseUrl = \"$URL\" |.backend.cors.origin = \"$URL\"" app-config.yaml
+              echo -n "."
+        
+            
+              cat << _EOF_ >> app-config-update.yaml
+              auth:
+                environment: development
+                providers:
+                  github:
+                    development:
+                      clientId: ${GITHUB__APP__CLIENT_ID}
+                      clientSecret: ${GITHUB__APP__CLIENT_SECRET}
+              catalog:
+                locations:
+                - target: ${DEVELOPER_HUB__CATALOG__URL}
+                  type: url
+                rules:
+                - allow:
+                  - Component
+                  - System
+                  - Group
+                  - Resource
+                  - Location
+                  - Template
+                  - API
+              integrations:
+                github:
+                - apps:
+                  - appId: ${GITHUB__APP__APP_ID}
+                    clientId: ${GITHUB__APP__CLIENT_ID}
+                    clientSecret: ${GITHUB__APP__CLIENT_SECRET}
+                    privateKey: |
+                      ${GITHUB__APP__PRIVATE_KEY}
+                    webhookSecret: ${GITHUB__APP__WEBHOOK_SECRET}
+                    webhookUrl: GITHUB__APP__WEBHOOK_URL
+                  host: github.com
+              techdocs:
+                builder: local
+                generator:
+                  runIn: local
+                publisher:
+                  type: local
+              _EOF_
+              echo -n "."
```

Comment:
That's another snippet I think we should consider moving to a standalone template file. And also, how the environment variables, like `${GITHUB__APP__PRIVATE_KEY}` are substituted?

Reply:
See my previous comment.